pandas/tests/indexes/period/test_indexing.py

```diff
@@ -310,3 +311,60 @@ def test_take_fill_value(self):
 
         with pytest.raises(IndexError):
             idx.take(np.array([1, -5]))
+
+    def test_truncate(self):
+        # GH 17717
+        idx1 = pd.PeriodIndex([
+            pd.Period('2017-09-02'),
+            pd.Period('2017-09-02'),
+            pd.Period('2017-09-03')
+        ])
+        series1 = pd.Series([1, 2, 3], index=idx1)
+        result1 = series1.truncate(after='2017-09-02')
```

Comment:
I would like to have a few period index specific tests that exercise the engine. you can look at what tests we have for example for DatetimeIndex / TimedeltaIndex for the same.